lib/portlayer/storage/volume.go

```diff
@@ -43,21 +67,35 @@ type Volume struct {
 
 	// Namespace in the storage layer to look up this volume.
 	SelfLink *url.URL
+
+	// Backing device
+	Device Disk
 }
 
-// VolumeStorer is an interface to create, remove, enumerate, and get Volumes.
-type VolumeStorer interface {
-	// Creates a volume on the given volume store, of the given size, with the given metadata.
-	VolumeCreate(ctx context.Context, ID string, store *url.URL, capacityMB uint64, info map[string][]byte) (*Volume, error)
+// NewVolume creates a Volume
+func NewVolume(store *url.URL, ID string, device Disk) (*Volume, error) {
+	storeName, err := util.VolumeStoreName(store)
+	if err != nil {
+		return nil, err
+	}
 
-	// Get an existing volume via it's ID.
-	VolumeGet(ctx context.Context, ID string) (*Volume, error)
+	selflink, err := util.VolumeURL(storeName, ID)
+	if err != nil {
+		return nil, err
+	}
 
-	// Destroys a volume
-	VolumeDestroy(ctx context.Context, ID string) error
+	// Set the label to the md5 of the ID
+	h := md5.New()
```

Comment:
Close.  Looks like it's 16 characters.

> If the optional argument new-label is present, then e2label will set the filesystem label to be new-label.  Ext2 filesystem labels can be at most 16 characters long; if new-label is longer than 16 characters, e2label will truncate it and print a warning message.

I'll update this with a comment referring to the `e2label` manpage.